### PR TITLE
Intercom construction and Deconstruction. 

### PR DESCRIPTION
**Pull request by SirBloodBuske**
I THINK I DID IT, this time.

### CODE REVIEW

**Comment by SirBloodBuske** [comment]
Adding to the recipe lists..

File: code/game/objects/items/mountable_frames/intercom.dm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/obj/item/mounted/frame/intercom
```
Comment (tigercat2000) [inline code/game/objects/items/mountable_frames/intercom.dm]: >uses mountable_frame system
GJ

File: code/game/objects/items/devices/radio/intercom.dm [hunk 2]
```diff
@@ -41,32 +56,107 @@
 			return -1
 	if (!src.listening)
 		return -1
-	if(freq in ANTAG_FREQS)
+	if(freq == SYND_FREQ)
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/radio/intercom.dm]: May want to look at the ramifications of changing this.

File: code/game/objects/items/devices/radio/intercom.dm [hunk 3]
```diff
@@ -41,32 +56,107 @@
...
 		if(!(src.syndie))
 			return -1//Prevents broadcast of messages over devices lacking the encryption
 
 	return canhear_range
 
 
-/obj/item/device/radio/intercom/hear_talk(mob/M as mob, msg)
-	if(!src.anyai && !(M in src.ai))
+/obj/item/device/radio/intercom/hear_talk(message, atom/movable/speaker, message_langs, raw_message, radio_freq)
+	if(!src.anyai && !(speaker in src.ai))
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/radio/intercom.dm]: May want to look at the ramifications of changing this. Reason being we have completely different saycode than VG's terribad saycode (which we killed off, entirely).

File: code/game/objects/items/devices/radio/intercom.dm [hunk 4]
```diff
@@ -5,30 +5,45 @@
 	anchored = 1
 	w_class = 4.0
 	canhear_range = 2
-	flags = CONDUCT | NOBLOODY
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/radio/intercom.dm]: Err, why?

**Comment by SirBloodBuske** [comment]
Yeahhh I butchered some VG code.. Ill fix it

File: code/game/objects/items/devices/radio/intercom.dm [hunk 5]
```diff
@@ -53,20 +69,95 @@
...
+			update_icon()
+			return
+		on = areaMaster.powered(EQUIP) // set "on" to the power status
+		update_icon()
 
-		if(!on)
-			icon_state = "intercom-p"
-		else
-			icon_state = "intercom"
+/obj/item/weapon/intercom_electronics
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/radio/intercom.dm]: These will be unobtainable without them being properly added elsewhere.

Comment (SirBloodBuske) [inline code/game/objects/items/devices/radio/intercom.dm]: I swear those are added elsewhere, or it is fixed proper lemme checks

**Pull request merged**